### PR TITLE
Stranded gamemode

### PR DESCRIPTION
**Pull request by caelaislinn**
Features:
- New "Stranded" gamemode. The players are UNSC survivors crash landed on an alien planet with waves of Flood coming in. Scavenge resources from around the map and build a base to hold them off, and survive until the pelican arrives. 

Ready for review but awaiting some final tweaks before merging.

### CODE REVIEW

File: code/modules/halo/flood/flood.dm [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+	//killing a spore can kill others nearby
+	/*for(var/mob/living/simple_animal/hostile/flood/infestor/S in view(1,src))
+		if(prob(33))
+			S.health = 0*/
+	return ..()
+
+/mob/living/simple_animal/hostile/flood/infestor/examine(mob/user, var/distance = -1, var/infix = "", var/suffix = "")
+	..()
+	if(swarm_size > 1)
+		user.visible_message("<span class='warning'>There is [swarm_size] in the swarm.</span>")
```
Comment (X0-11) [inline code/modules/halo/flood/flood.dm]: I think you want to_chat(user,message) here. Visible_message shows a message to people around them too, I think

File: code/modules/halo/gamemodes/stranded/turfs_areas.dm [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	var/valid_tool
+	for(var/valid_type in usable_tools)
+		if(istype(W,valid_type))
+			valid_tool = 1
+			break
+
+	if(valid_tool)
+		if (dug)
+			user.show_message("\red This area has already been dug")
```
Comment (X0-11) [inline code/modules/halo/gamemodes/stranded/turfs_areas.dm]: to_chat(user,message) here too. I think travis dislikes the \red's, switch to standard "<span class = "...

File: code/modules/halo/gamemodes/stranded/turfs_areas.dm [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			return
+
+		var/turf/T = user.loc
+		if (!(istype(T)))
+			return
+
+		user.show_message("\red You start digging.")
```
Comment (X0-11) [inline code/modules/halo/gamemodes/stranded/turfs_areas.dm]: Here too

File: code/modules/halo/gamemodes/stranded/turfs_areas.dm [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+		playsound(user.loc, 'sound/effects/rustle1.ogg', 50, 1)
+
+		if(!do_after(user,40)) return
+
+		user.show_message("\blue You dug a hole.")
```
Comment (X0-11) [inline code/modules/halo/gamemodes/stranded/turfs_areas.dm]: And here

File: code/modules/materials/material_recipes.dm [hunk 5]
```diff
@@ -7,28 +7,28 @@
 	recipes = list()
 
 	// If is_brittle() returns true, these are only good for a single strike.
-	recipes += new/datum/stack_recipe("[display_name] baseball bat", /obj/item/weapon/material/twohanded/baseballbat, 10, time = 20, one_per_turf = 0, on_floor = 1, supplied_material = "[name]")
-	recipes += new/datum/stack_recipe("[display_name] ashtray", /obj/item/weapon/material/ashtray, 2, one_per_turf = 1, on_floor = 1, supplied_material = "[name]")
-	recipes += new/datum/stack_recipe("[display_name] spoon", /obj/item/weapon/material/kitchen/utensil/spoon/plastic, 1, on_floor = 1, supplied_material = "[name]")
-	recipes += new/datum/stack_recipe("[display_name] ring", /obj/item/clothing/ring/material, 1, on_floor = 1, supplied_material = "[name]")
+	//recipes += new/datum/stack_recipe("[display_name] baseball bat", /obj/item/weapon/material/twohanded/baseballbat, 10, time = 20, one_per_turf = 0, on_floor = 1, supplied_material = "[name]")
```
Comment (X0-11) [inline code/modules/materials/material_recipes.dm]: I don't think this is the best way to do this. Making a new material with a slightly different name and custom recipes would be better.

File: code/modules/halo/gamemodes/stranded/misc.dm [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+		damtype = BRUTE
+		force = 10
+
+/obj/item/device/flashlight/flaming_torch/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
+	if(on)
+		fuel_ticks_left -= 10
+
+/obj/item/device/flashlight/flaming_torch/examine(mob/user, var/distance = -1, var/infix = "", var/suffix = "")
+	..()
+	user.show_message("It has [fuel_ticks_left] out of [max_fuel_ticks] units of fuel left ([100 * fuel_ticks_left/max_fuel_ticks]%).")
```
Comment (X0-11) [inline code/modules/halo/gamemodes/stranded/misc.dm]: to_chat(user,message)

File: code/modules/halo/gamemodes/stranded/flora.dm [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+		spawn(0)
+			if(do_after(user, 20 + 20 / chopping))
+				woodleft -= 1
+				user.show_message("<span class='info'>You chop some wood from [src]. There is [src.woodleft] planks left.</span>")
+				var/obj/item/stack/material/wood/W = new(src.loc)
+				W.amount = 10
+				if(woodleft <= 0)
+					qdel(src)
+	else
+		user.show_message("<span class='warning'>You need something sharp to chop down [src]</span>")
```
Comment (X0-11) [inline code/modules/halo/gamemodes/stranded/flora.dm]: to_chat(user,message)

File: code/modules/halo/flood/flood.dm [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+	melee_damage_upper = 15
+
+/mob/living/simple_animal/hostile/flood/carrier/AttackingTarget()
+	if(!Adjacent(target_mob))
+		return
+
+	health = 0
+
+/mob/living/simple_animal/hostile/flood/carrier/death(gibbed, deathmessage = "bursts!")
+	src.show_message("<span class='danger'>[src] bursts, propelling flood spores in all directions!</span>")
```
Comment (X0-11) [inline code/modules/halo/flood/flood.dm]: visible_message()

**Review by X0-11 — CHANGES_REQUESTED** [review]

File: html/changelogs/cael_stranded.yml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - wip: "Adds Stranded gamemode. Your UNSC ship has crashed landed on an abandoned alien world and now must fend off waves of Flood until your evacuation Pelican arrives!"
+  - wip: "Scavenge resources from your environment and build a base to hold them off."
+  - maptweak: "Stranded desert map"
+  - rscadd: "Adds new buildable structures: tank trap, barbed wire, sandbags"
+  - tweak: "Adds an atom despawner (testing)"
+  - wips: "Adds an experiement ambient lighting system (testing)"
```
Comment (X0-11) [inline html/changelogs/cael_stranded.yml]: Travis doesn't like this last one, I assume it's meant to be wip?

**Comment by X0-11** [comment]
It'd be better to have the entire stranded folder in /maps. Travis'll run some map-specific checks then.

File: baystation12.dme [hunk 10]
```diff
@@ -244,14 +244,8 @@
...
 #include "code\datums\observation\~cleanup.dm"
 #include "code\datums\outfits\_defines.dm"
-#include "code\datums\outfits\horror_killers.dm"
-#include "code\datums\outfits\misc.dm"
-#include "code\datums\outfits\nanotrasen.dm"
 #include "code\datums\outfits\outfit.dm"
-#include "code\datums\outfits\pirates.dm"
 #include "code\datums\outfits\spec_op.dm"
-#include "code\datums\outfits\tournament.dm"
-#include "code\datums\outfits\wizardry.dm"
```
Comment (bloxgate) [inline baystation12.dme]: 😞 No more spawning in wizards in full gear to mess with people?

File: code/game/machinery/floodlight.dm [hunk 11]
```diff
@@ -16,6 +16,9 @@
 	cell = new/obj/item/weapon/cell/crap(src)
 	..()
 
+	if(on)
+		turn_on()
```
Comment (bloxgate) [inline code/game/machinery/floodlight.dm:20]: Are these supposed to both be on or should one be off?

**Review by bloxgate — COMMENTED** [review]
I only really skimmed this, but I think @X0-11 got the majority of any problems there should be. Once those issues are fixed, I'll approve as well.

**Comment by caelaislinn** [comment]
Thanks for the pickups XO. I was pretty tired and didn't realise how many chat log muckups I made

File: code/modules/materials/material_recipes.dm (hunk 5 above)
Comment (caelaislinn) [reply code/modules/materials/material_recipes.dm]: I want to cut out a lot of the bloat, particularly RP stuff like this

Comment (X0-11) [reply code/modules/materials/material_recipes.dm]: It'd be better to leave them commented then, just in case they're needed for future reference.

**Comment by X0-11** [comment]
How is this gamemode supposed to end? I don't see it using the built in check_finished proc to auto-end. Am I missing something?

File: code/game/machinery/floodlight.dm (hunk 11 above)
Comment (caelaislinn) [reply code/game/machinery/floodlight.dm:20]: Yes. This is ran when the object is created and if on == 1 at that time then it activates the floodlight

**Comment by caelaislinn** [comment]
I missed that too. Fixing that now

**Comment by caelaislinn** [comment]
#289

**Comment by bloxgate** [comment]
You have an unclosed \<span> somewhere in flood.dm:
./code/modules/halo/flood/flood.dm

	<span> - Excess of 1 opening tag(s)

		Line 173

**Review by bloxgate — CHANGES_REQUESTED** [review]
It looks like something you've done recently has broken a lot of stuff on all the map tests. It appears that a lot of stuff now fails to create objects due to null locations, or other errors.

File: code/modules/halo/flood/flood_npc.dm [hunk 12]
```diff
@@ -0,0 +1,9 @@
+
+/obj/effect/landmark/flood_spawn
+	name = "flood spawn marker"
+
+/obj/effect/landmark/flood_spawn
```
Comment (X0-11) [inline code/modules/halo/flood/flood_npc.dm]: Duplicate definition?

File: code/modules/halo/gamemodes/stranded/evac_pelican.dm [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+		</span>")
+	//qdel(I)
+
+/obj/structure/evac_pelican/attack_hand(var/mob/M)
+	attempt_enter(M)
+
+/obj/structure/evac_pelican/proc/attempt_enter(var/mob/L)
+	if(isliving(usr))
+		var/mob/living/M = usr
+		if(M.stat == CONSCIOUS)
```
Comment (X0-11) [inline code/modules/halo/gamemodes/stranded/evac_pelican.dm]: !incapacitated() might be better here.

File: code/modules/halo/gamemodes/stranded/structures.dm [hunk 14]
```diff
@@ -0,0 +1,393 @@
...
+/obj/item/empty_sandbags/examine(mob/user)
+	to_chat(user,"It is oriented [src.dir].")
+
+/obj/item/empty_sandbags/verb/arrange_dir()
+	set name = "Choose sandbag direction"
+	set category = "Object"
+	set src = view(1)
+
+	var/list/dir_options = list("North" = NORTH,"South" = SOUTH, "East" = EAST, "West" = WEST)
+	var/newdirname = input("Pick new direction for this barbed wire","New wire direction","North") in dir_options
```
Comment (X0-11) [inline code/modules/halo/gamemodes/stranded/structures.dm]: Calling it barbed wire, not sandbags.

**Review by X0-11 — APPROVED** [review]
Three changes, but overall looks fine.

**Pull request merged**